html/befuncs/snips.php

```diff
@@ -28,9 +29,23 @@ function genNavBar(){
 		<nav id="main_nav">
 			<div>
 				<?php
+					if(array_key_exists('redirect_to',$_GET) && array_key_exists('signature',$_GET)){
+						$params = http_build_query(array(
+								'redirect_to' => $_GET['redirect_to'],
+								'signature' => $_GET['signature'],
+							));
+					}else{
+						$crypt = new crypt();
```

Comment:
why would we even need all this manual cryptography at all? Can't we just add the url in plaintext?

Reply:
A bad actor could send you a malicious `redirect_to` parameter.

Reply:
how

Reply:
Nvm just `host` is enough in this case, eg `?redirect_to=google.com` would redirect you to `google` (could also be a more malicious site of course). But there are also other possibilities like redirect you to a "delete user page".

Reply:
…so? In that case, we just need to keep the redirect on our site, so making the redirects relative to the currect domain would suffice.

e.g. `?redirect_to=/music/`

seems smarter to me than encrypting the url parameter and doing all this extra effort to… prevent arbitrary redirects.

Reply:
I understand you've done all this effort and it sucks to delete like 75% of this PR, but I think it's the best option going forward.

Reply:
Imo just signing the url is more reliable than url checking/parsing, but I can rewrite it if you don't want to use cryptography.

Reply:
I try to keep the site as simple as possible, so yeah please